test/unit/local/steps/incomplete_fixer.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-env mocha */
+/* @flow */
+
+const _ = require('lodash')
+const path = require('path')
+
+const should = require('should')
+const Builders = require('../../../support/builders')
+const configHelpers = require('../../../support/helpers/config')
+const pouchHelpers = require('../../../support/helpers/pouch')
+const { ContextDir } = require('../../../support/helpers/context_dir')
+
+const metadata = require('../../../../core/metadata')
+const Buffer = require('../../../../core/local/steps/buffer')
+const incompleteFixer = require('../../../../core/local/steps/incomplete_fixer')
+
+const completedEvent = event =>
+  _.pick(event, ['_id', 'path', 'kind', 'md5sum', 'action', 'incomplete'])
+
+const completionChanges = async (buffer) => await buffer.pop().map(completedEvent)
+
+describe('local/steps/initialDiff', () => {
+  let builders
+  let syncDir
+  let buffer
+  let options
+
+  const CHECKSUM = 'checksum'
+  const checksumer = {
+    push: path => CHECKSUM,
+    kill: () => {}
+  }
+
+  before('instanciate config', configHelpers.createConfig)
+  before(function () {
+    options = { syncPath: this.config.syncPath, checksumer }
+    syncDir = new ContextDir(this.config.syncPath)
+  })
+  beforeEach('instanciate pouch', pouchHelpers.createDatabase)
+  beforeEach('populate pouch with documents', function () {
+    builders = new Builders({pouch: this.pouch})
+    buffer = new Buffer()
+  })
+  afterEach('clean pouch', pouchHelpers.cleanDatabase)
+  after('clean config directory', configHelpers.cleanConfig)
+
+  context('without any complete "renamed" event', () => {
+    it('drops incomplete events', async function () {
+      const events = [
+        builders.event().incomplete().action('created').path('foo1').build(),
+        builders.event().incomplete().action('modified').path('foo2').build(),
+        builders.event().incomplete().action('deleted').path('foo3').build(),
+        builders.event().incomplete().action('scan').path('foo4').build(),
+        builders.event().action('initial-scan-done').path('').build()
+      ]
+      buffer.push(_.cloneDeep(events))
+
+      const completedBuffer = await incompleteFixer.loop(buffer, options)
+      should(await completedBuffer.pop()).deepEqual([
+        events[events.length - 1]
```

Comment:
Maybe extract the single expected event to a separate `const`, so we don't rely on position?

```js
const initialScanDone = ...
const events = [
  builder.event().whatever().build(),
  builder.event().whatever().build(),
  initialScanDone
]
...
should(...).deepEqual([
  initialScanDonw
])
```